tests/test_exception.c

```diff
@@ -329,4 +329,26 @@ void test_exception(TestContext *ctx) {
 #undef ASSERT_REG_GP
 #undef ASSERT_REG_FP_HANDLER
 #undef ASSERT_REG_GP_HANDLER
-#undef ASSERT_REG
\ No newline at end of file
+#undef ASSERT_REG
+
+
+static volatile bool tsh_called = true;
+static volatile uint32_t tsh_code = 0;
+
+void test_syscall_handler(exception_t *exc, uint32_t code) {
```

Comment:
Is there a reason why we don't use the gcc extension here? Is the stack destroyed or something?

Reply:
you mean the nested function? No there isn't a good reason, I could do that. I figured out that, if for any reason (= a bug in libdragon) the syscall handler remains registered to a nested function, it might lead to a hard-to-diagnose crash if it gets invoked, as the stack would have been destroyed.